mesonbuild/dependencies/base.py

```diff
@@ -420,13 +420,14 @@ def get_partial_dependency(self, *, compile_args: bool = False,
         new = copy.copy(self)
         new._id = uuid.uuid4().int
         if not compile_args:
-            new.compile_args = []
+            if includes:
+                new.compile_args = [c for c in self.compile_args if c.startswith('-I')]
```

Comment:
That's not sufficient, at the very least this needs to be `c.startswith(('-I', '/I', '-isystem', '-idir-after'))`, but even that probably isn't sufficient.

I'm also really, really, hesitant to do this in general, since the only valid case where a we could get a raw include is from pkg-config (or maybe CMake?). I'd tried to fix this previously and ran into pkgconf bugs on mingw: https://github.com/mesonbuild/meson/pull/10122

At the very least I think this should be limited to pkg-config dependencies.

Reply:
I was actually thinking about that PR this weekend

Reply:
I see... But OTOH, it will not change the behavior if `compile_args` is `true`. The only thing it changes is when you request `includes` but not `compile_args`. 

Not sure about `-isystem` and `-idirafter`. Should includes from a dependency include system dirs? For `/I`, I though it was already replaced by `-I` by meson, but I may be wrong... The only thing I know is that `pkgconf --cflags-only-I` will only keep arguments beginning with `-I`.

Reply:
I proposed this because I get caught this morning, trying to use the includes from a dependency. I forgot it was only working for internal dependencies...

Reply:
My concern is it changes the behavior of a `declare_dependency()`
```meson
x = declare_dependency(compile_args : ['-Dfoo', '-Ibar']).partial_dependency(include_directories : true)
```
Currently gives an empty declare_dependency. This is what we intend, we don't support putting `-I` flags in compile_args, you get to keep the pieces if you do that.

Reply:
This wouldn't change, since my changes only apply to external dependencies. Behavior of internal_dependencies remains the same.

Reply:
hmmm, I'm still not sure. I'll have a look at what ends up here and then go from there.

Either way a test would be good. There might be something in that PR I linked you could steal.